File: frontend-vue/src/services/shop/product/CartService.js
import http from '../../../utils/http-common'

class CartService{
    // 장바구니 전체 조회
    getAll(title,page,size){
        return http.get(`/shop/cart?title=${title}&page=${page}&size=${size}`);
    }

    // 장바구니 저장
    save(cart){
        return http.post("/shop/cart", cart);
    }
}

export default new CartService();

File: SimpleDms/src/main/java/org/example/simpledms/service/shop/product/CartService.java
package org.example.simpledms.service.shop.product;

import org.example.simpledms.model.entity.shop.product.Cart;
import org.example.simpledms.model.entity.shop.product.Product;
import org.example.simpledms.repository.shop.product.CartRepository;
import org.example.simpledms.repository.shop.product.ProductRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

/**
 * packageName : org.example.simpledms.service.shop.product
 * fileName : CartService
 * author : PC
 * date : 2024-04-10
 * description :
 * 요약 :
 * <p>
 * ===========================================================
 * DATE            AUTHOR             NOTE
 * -----------------------------------------------------------
 * 2024-04-10         PC          최초 생성
 */
@Service
public class CartService {

    private final CartRepository cartRepository;

    @Autowired
    public CartService(CartRepository cartRepository) {
        this.cartRepository = cartRepository;
    }


    public Cart save(Cart cart){
        Cart cart1 = cartRepository.save(cart);
        return cart1;
    }
}
